src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -325,18 +325,6 @@ public void Track(ITelemetry telemetry)
             // It is hidden from intellisense to prevent customer confusion.
             if (this.IsEnabled())
             {
-                string instrumentationKey = this.Context.InstrumentationKey;
-
-                if (string.IsNullOrEmpty(instrumentationKey))
-                {
-                    instrumentationKey = this.configuration.InstrumentationKey;
-                }
-
-                if (string.IsNullOrEmpty(instrumentationKey))
-                {
-                    return;
```

Comment:
Is that done this way instead of checking telemetry item because we assume that IKey can be set by telemetry initializer?